### ISSUE TITLE
Option to persist writeOutputsToSecret secret

### ISSUE THREAD

**Issue by grrywlsn** [weaveworks/tf-controller#538]
TLDR: tf-controller deletes any secret specified in the `writeOutputsToSecret` value, even if the secret was created by something else. It'd be great to have a way to make this optional, so the secret might not be deleted too.

I've got a helm chart that is setting up some terraform resources on install (with helm hooks), and the outputs of that terraform execution is some immutable config I want to persist in a secret. However, once helm has finished installed and is cleaning up any hooked resources, when it deletes the terraform custom resources that cascades and deletes the secret. 

I hoped adding the target secret as a separate helm resource would avoid this, but that doesn't stop it being deleted.

imo if tf-controller didn't create the secret, it feels like the default should _not_ be to delete it on cleanup; however a simple boolean to make the behaviour clean would be a great addition for this problem.

**Comment by chanwit** [comment]
Hi @grrywlsn 
Thank you for reporting this.

It's an interesting issue. Yep leaving secrets there in the cluster after the object is deleted would be an OK approach.
what do you think @Nalum @tomhuang12 ?

**Comment by tomhuang12** [comment]
Yeah I think we can add something like `.spec.writeOutputsToSecret.destroyResourcesOnDeletion`, defaulting to true.

**Comment by Nalum** [comment]
Sounds good to me :+1:

**Comment by squaremo** [comment]
> Yeah I think we can add something like `.spec.writeOutputsToSecret.destroyResourcesOnDeletion`, defaulting to true.

For the sake of backward compatibility, it's better to make additional flags default to `false`. I would suggest `keepResourcesOnDeletion`.

**Comment by LappleApple** [comment]
Moving conversation to #937 and closing.
